webapp/src/Controller/API/SubmissionController.php

```diff
@@ -386,6 +384,22 @@ protected function getQueryBuilder(Request $request): QueryBuilder
             $queryBuilder->andWhere('s.submittime < c.endtime');
         }
 
+        if (!$this->dj->checkrole('api_reader') &&
+            !$this->dj->checkrole('judgehost'))
+        {
+            $queryBuilder
+                ->join('s.team', 't')
```

Comment:
Above we should probably already join on team and enforce `t.enabled = 1` in general, because we consider a disabled team as deleted if I'm not mistaken.